### PR TITLE
enable replacing var (locale - PO files)

### PR DESCRIPTION
**Pull request by JaskRendix**
PR enables replacing variables inside the PO files. It uses the same method:

eg `${{name}} is running away!` gives back `Richard is running away!`

this will help to pass data without being forced to hardcode, especially numbers as well as names, etc. 

Example:
eg `you chose nr. {{var:xyz}} back then!` gives back `you chose nr. 3 back then!`

black, isort, tested, no new typehints

### CODE REVIEW

**Comment by Grimmys** [comment]
Wooooooooow!

It's funny because I was talking about something really similar a few weeks ago in the French translation channel: being able to put tokens (variables) into translation files is neat, because then we can avoid having to repeat the same translation for a keyword element of the game, since this repetitions could to lead to inconsistent translations of the same term (ex: _Tuxecenter_ has been recently translated as _Tuxecentre_ in French, but without such feature, in a few months, we could see a sentence using another variant like _Centre Tuxemon_).

TL;DR: Thank you a lot for your investment on this!

**Review by Grimmys — APPROVED** [review]

**Review by bitcraft — APPROVED** [review]
